src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -599,6 +678,78 @@ private static ImmutableArray<RefKind> GetRefKindsOrNull(ArrayBuilder<RefKind> r
             }
         }
 
+        // This fills in the arguments and parameters arrays in evaluation order.
+        private static void BuildExplicitArgumentsInEvaluationOrder(
+            SyntaxNode syntax,
+            Symbol methodOrIndexer,
+            bool expanded,
+            ImmutableArray<int> argsToParamsOpt,
+            ImmutableArray<BoundExpression> arguments,
+            CSharpCompilation compilation,
+            /* out */ ArrayBuilder<BoundExpression> argumentsBuilder,
+            /* out */ ArrayBuilder<ArgumentKind> argumentKindsBuilder,
+            /* out */ ArrayBuilder<ParameterSymbol> parametersBuilder,
+            /* out */ ArrayBuilder<ParameterSymbol> missingParametersBuilder)
+        {
+            ImmutableArray<ParameterSymbol> parameters = methodOrIndexer.GetParameters();
+            PooledHashSet<int> processedPeremeters = PooledHashSet<int>.GetInstance();
```

Comment:
Would `processedParameters` be simpler as `ArrayBuilder<bool>` of size `parameters.Length`?